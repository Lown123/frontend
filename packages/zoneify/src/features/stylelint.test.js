const { addDependency, addPackageJsonChange } = require('../install')
const { name, questions, run } = require('./stylelint')

jest.mock('../install')

describe('features/stylelint', () => {
  it('should have a name', () => {
    expect(name).toMatchSnapshot()
  })

  it('should have no questions', () => {
    expect(questions).toMatchSnapshot()
  })

  describe('run', () => {
    const testPackageJson = { name: 'test package' }

    run()

    it('should have a run function', () => {
      expect(typeof run).toBe('function')
    })

    it('should add a dependency', () => {
      expect(addDependency).toBeCalledWith({
        name: '@zonedigital/stylelint-config-zone',
        includePeerDependencies: true,
      })
    })

    it('should add a config change', () => {
      expect(addPackageJsonChange).toMatchSnapshot()
      expect(typeof addPackageJsonChange.mock.results[0].value).toBe('function')
      expect(addPackageJsonChange.mock.results[0].value(testPackageJson)).toMatchSnapshot()
    })
  })
})
