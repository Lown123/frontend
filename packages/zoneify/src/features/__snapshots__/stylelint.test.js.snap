// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/stylelint run should add a config change 1`] = `
[MockFunction addPackageJsonChange] {
  "calls": Array [
    Array [
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": [Function],
    },
  ],
}
`;

exports[`features/stylelint run should add a config change 2`] = `
Object {
  "name": "test package",
  "stylelint": Object {
    "extends": "@zonedigital/stylelint-config-zone",
  },
}
`;

exports[`features/stylelint should have a name 1`] = `"stylelint"`;

exports[`features/stylelint should have no questions 1`] = `undefined`;
